src/Swift.Bindings/src/Model/Provenance.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Represents the origin of a particular function
+/// </summary>
+public class Provenance {
```

Comment:
Isn't this the same property as `public required BaseDecl? ParentDecl { get; set; }`?

Reply:
BaseDecl, IIRC is a C# representation not a Swift declaration. Further, extensions make this more complicated because you can have multiple extension on the same type from difference modules.